factory/tests/factory_tests.rs

```diff
@@ -0,0 +1,112 @@
+use factory_io::*;
+use gstd::{prelude::*, ActorId};
+use gtest::System;
+mod utils;
+use utils::*;
+
+#[test]
+fn fee_to() {
+    let sys = System::new();
+    init_factory(&sys);
+    let factory = sys.get_program(1);
+    _ = set_fee_to_utils(&factory, FEE_SETTER, ActorId::from(NEW_FEE_TO));
+    let res = fee_to_utils(&factory, FEE_SETTER);
+    let message = FactoryEvent::FeeTo {
+        address: ActorId::from(NEW_FEE_TO),
+    }
+    .encode();
+    assert!(res.contains(&(FEE_SETTER, message)));
+}
+
+#[test]
+fn set_fee_to() {
+    let sys = System::new();
+    init_factory(&sys);
+    let factory = sys.get_program(1);
+    // should be sent by a fee_setter
+    let res = set_fee_to_utils(&factory, FEE_SETTER, ActorId::from(NEW_FEE_TO));
+    let message = FactoryEvent::FeeToSet {
+        fee_to: ActorId::from(NEW_FEE_TO),
+    }
+    .encode();
+    assert!(res.contains(&(FEE_SETTER, message)));
+
+    // check if new fee_to is in state
+    check_fee_to(&factory, ActorId::from(NEW_FEE_TO));
+}
+
+#[test]
+fn set_fee_to_failures() {
+    let sys = System::new();
+    init_factory(&sys);
+    let factory = sys.get_program(1);
+    // MUST fail since the USER is not a fee setter
+    let res = set_fee_to_utils(&factory, USER, ActorId::from(NEW_FEE_TO));
```

Comment:
Here and everywhere.
```suggestion
    let res = utils::set_fee_to(&factory, USER, ActorId::from(NEW_FEE_TO));
```